pkg/namespace/consts.go

```diff
@@ -59,5 +60,8 @@ var (
 	TailPaddingNamespace = MustNewV0([]byte{0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFE})
 
 	// ParitySharesNamespace is the namespace reserved for erasure coded data.
-	ParitySharesNamespace = MustNewV0([]byte{0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF})
+	ParitySharesNamespace = Namespace{
+		Version: math.MaxUint8,
+		ID:      bytes.Repeat([]byte{0xFF}, NamespaceIDSize),
+	}
```

Comment:
This seems adjacent to the issue you've outlined and I'm wondering the motivation for this being the max version

Reply:
The motivation is that a parity namespace with version 0 isn't the largest possible 33 byte namespace. A parity namespace with max version is the largest possible 33 byte namespace so it is ignored by the [IgnoreMaxNamespace](https://github.com/celestiaorg/nmt/blob/89434babb8f93c49f87859de909ab63123abdcef/docs/nmt-lib.md?plain=1#L18) option. celestia-node assumes that each row root is an NMT root with this option enabled.

Additional Context in Slack: https://celestia-team.slack.com/archives/C02PFFCGMNW/p1680900937688359

Reply:
Yeah that makes sense. Was trying to reason if the parity share would have any need to change the version